src/components/Employee/Deductions/DeductionsForm/ChildSupportForm.tsx

```diff
@@ -0,0 +1,209 @@
+
+const ChildSupportPaymentPeriodSchema = z.nativeEnum(PaymentPeriod)
+const ChildSupportSchema = z.object({
+  state: z.string(),
+  fipsCode: z.string(),
+  caseNumber: z.string(),
+  amount: z.number().min(0).transform(String),
+  payPeriodMaximum: z
+    .number()
+    .min(0)
+    .transform(val => (val > 0 ? val.toString() : null))
+    .nullable(),
+  paymentPeriod: ChildSupportPaymentPeriodSchema,
+})
+
+export type ChildSupportInputs = z.input<typeof ChildSupportSchema>
+export type ChildSupportPayload = z.output<typeof ChildSupportSchema>
+
+interface ChildSupportFormProps extends CommonComponentInterface<'Employee.Deductions'> {
+  employeeId: string
+  deduction?: Garnishment | null
+  handleStateAgencySelect: (stateAgency: string) => void
+  csAgencies: { label: string; value: string }[]
+  counties: { label: string; value: string }[]
+}
+
+function ChildSupportForm({
+  deduction,
+  handleStateAgencySelect,
+  csAgencies,
+  counties,
+  employeeId,
+}: ChildSupportFormProps) {
+  const { onEvent, baseSubmitHandler } = useBase()
+  const { t } = useTranslation('Employee.Deductions')
+  const Components = useComponentContext()
+
+  const { mutateAsync: createDeduction, isPending: isPendingCreate } =
+    useGarnishmentsCreateMutation()
+  const { mutateAsync: updateDeduction, isPending: isPendingUpdate } =
+    useGarnishmentsUpdateMutation()
+  const isPending = isPendingCreate || isPendingUpdate
+
+  const defaultChildSupportValues: ChildSupportInputs = useMemo(() => {
+    return {
+      amount: deduction?.amount ? Number(deduction.amount) : 0,
+      payPeriodMaximum: deduction?.payPeriodMaximum ? Number(deduction.payPeriodMaximum) : null,
+      state: deduction?.childSupport?.state || '',
+      fipsCode: deduction?.childSupport?.fipsCode || '',
+      caseNumber: deduction?.childSupport?.caseNumber || '',
+      paymentPeriod: deduction?.childSupport?.paymentPeriod as PaymentPeriod,
+    }
+  }, [deduction])
+
+  const childSupportFormMethods = useForm<ChildSupportInputs, unknown, ChildSupportPayload>({
+    resolver: zodResolver(ChildSupportSchema),
+    defaultValues: defaultChildSupportValues,
+  })
+  const { reset: resetChildSupportForm, control } = childSupportFormMethods
+  const watchedStateAgency = useWatch({ control, name: 'state' })
+
+  useEffect(() => {
+    resetChildSupportForm(defaultChildSupportValues)
+  }, [deduction, defaultChildSupportValues, resetChildSupportForm])
+
+  const onChildSupportSubmit: SubmitHandler<ChildSupportPayload> = async data => {
+    const childSupport = {
+      state: data.state,
+      paymentPeriod: data.paymentPeriod,
+      fipsCode: data.fipsCode,
+      caseNumber: data.caseNumber,
+    }
+
+    await baseSubmitHandler(data, async payload => {
+      const requestBody = {
+        active: true,
+        amount: payload.amount,
+        description: `Child Support - ${childSupport.caseNumber}`, // child support description follows prefix type + case number convention
```

Comment:
Hard coding here feels sus, might be worth constantitizing this somehow

Reply:
it is odd and technically description is optional, I was just following how the payload gets formatted via existing gws-flows for deductions